### PR TITLE
Add allow non root login option

### PR DESCRIPTION
**Pull request by baurine**
desc: //TODO

Wait PR #985 to be merged.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -48,6 +49,7 @@ func Default() *Config {
 		TiDBTLSConfig:      nil,
 		EnableTelemetry:    true,
 		EnableExperimental: false,
+		AllowNonRootLogin:  false,
```
[the remark this answers is not in the document]
Comment (baurine) [reply pkg/config/config.go]: Done.

File: Makefile [hunk 2]
```diff
@@ -46,4 +46,4 @@ endif
 	go build -o bin/tidb-dashboard -ldflags '$(LDFLAGS)' -tags "${BUILD_TAGS}" cmd/tidb-dashboard/main.go
 
 run:
-	bin/tidb-dashboard --debug --experimental
+	bin/tidb-dashboard --debug --experimental --allow-non-root-login
```
Comment (breeswish) [inline Makefile:49]: As it is by default, this is no longer needed.

File: cmd/tidb-dashboard/main.go [hunk 3]
```diff
@@ -73,6 +73,7 @@ func NewCLIConfig() *DashboardCLIConfig {
 	flag.StringVar(&cfg.CoreConfig.PDEndPoint, "pd", cfg.CoreConfig.PDEndPoint, "PD endpoint address that Dashboard Server connects to")
 	flag.BoolVar(&cfg.CoreConfig.EnableTelemetry, "telemetry", cfg.CoreConfig.EnableTelemetry, "allow telemetry")
 	flag.BoolVar(&cfg.CoreConfig.EnableExperimental, "experimental", cfg.CoreConfig.EnableExperimental, "allow experimental features")
+	flag.BoolVar(&cfg.CoreConfig.AllowNonRootLogin, "allow-non-root-login", cfg.CoreConfig.AllowNonRootLogin, "allow non root sql user login")
```
Comment (breeswish) [inline cmd/tidb-dashboard/main.go:76]: How about calling it EnableNonRootLogin to be unified with the rest?

**Review by breeswish — COMMENTED** [review]
I think you can merge this PR with #985 as it changes very few. I'm fine with the rest.

**Comment by baurine** [comment]
Close it cause it is merged to PR #985

File: cmd/tidb-dashboard/main.go (hunk 3 above)
Comment (baurine) [reply cmd/tidb-dashboard/main.go:76]: Done

File: Makefile (hunk 2 above)
Comment (baurine) [reply Makefile:49]: Done

**Pull request closed without merging**